doc/modules/computing.rst

```diff
@@ -504,20 +504,138 @@ Links
+that increasing the number of jobs is always a good thing. In some cases it
+can be highly detrimental to performance to run multiple copies of some
+estimators or functions in parallel (see oversubscription below).
+
+The nested use of joblib-based parallelism with the same backend will result
+in an exception. For example, ``GridSearchCV(OneVsRestClassifier(SVC(),
+n_jobs=2), n_jobs=2)`` won't work.
+
+OpenMP-based parallelism
+........................
+
+OpenMP is used to parallelize code written in Cython or C, relying on
+multi-threading exclusively. By default (and unless joblib is trying to
+avoid oversubscription), the implementation will use as many threads as
+possible.
+
+You can control the exact number of threads that are used via the
+``OMP_NUM_THREADS`` environment variable::
+
+    OMP_NUM_THREADS=4 python my_script.py
+
+Parallel Numpy routines from numerical libraries
+................................................
+
+Scikit-learn relies heavily on Numpy, which may be configured to use a
+multi-threaded numerical processor library like MKL or OpenBLAS.
+
+The number of threads used by the BLAS or MKL libraries can be set via the
+``MKL_NUM_THREADS`` and ``OPENBLAS_NUM_THREADS`` environment variables.
+
+Please note that scikit-learn has no direct control over these
+implementations. Scikit-learn solely relies on Numpy.
+
+Oversubscription: spawning too many threads
+...........................................
+
+It is generally recommended to avoid using significantly more processes or
+threads than the number of CPUs on a machine. Over-subscription happens when
+you're trying to spawn too many threads at the same time.
+
+Suppose you have a machine with 4 CPUs. Consider a case where you're running
+a :class:`~GridSearchCV` (parallelized with joblib) with ``n_jobs=8`` over
+a :class:`~HistGradientBoostingClassifier` (parallelized with OpenMP). Each
+instance of :class:`~HistGradientBoostingClassifier` will spawn 4 threads
+(since you have 4 cores). That's a total of ``8 * 4 = 32`` threads, which
+leads to oversubscription of physical CPU resources and to scheduling
+overhead.
+
+Oversubscription can arise in the exact same fashion with parallelized
+routines from MKL or OpenBLAS that are nested in joblib calls.
+
+Starting from ``joblib >= 0.14``, when the ``loky`` backend is used (which
+is the default), joblib will tell its child **processes** to limit the
+number of threads they can use, so as to avoid oversubscription. Back to our
+example from above, that means that if the joblib call from
+:class:`~GridSearchCV` spawns processes, each process will only be able to
+use 1 thread instead of 4, thus mitigating the oversubscription issue.
+
+Note that:
+
+- Manually setting one of the environment variables (``OMP_NUM_THREADS``,
+  ``MKL_NUM_THREADS``, or ``OPENBLAS_NUM_THREADS``) will take precedence
+  over what joblib tries to do. You can also use a context manager (see
+  joblib docs linked below)
```

Comment:
You can add that this value will be the max number of threads of these libraries in each process. Meaning total number of threads = n_jobs * (OMP/MKL/OPENBLAS)_NUM_THREADS